hashicorp/vault
# Backport of Secrets Sync: only call destinations and associations if isActivated into release/1.16.x

## Backport

This PR is auto-generated from #26299 to be assessed for backporting due to the inclusion of the label backport/1.16.x.


:rotating_light:
>**Warning** automatic cherry-pick of commits failed. If the first commit failed,
you will see a blank no-op commit below. If at least one commit succeeded, you
will see the cherry-picked commits up to, _not including_, the commit where
the merge conflict occurred.

The person who merged in the original PR is:
@Monkeychip
This person should manually cherry-pick the original PR into a new backport PR,
and close this one when the manual backport PR is merged in.

> merge conflict error: POST https://api.github.com/repos/hashicorp/vault/merges: 409 Merge conflict []



The below text is copied from the body of the original PR.

---

During the opt-in banner work we missed a conditional to not call the associations and destinations endpoints if the feature has not been activated. This PR moves the `isActivated` definition to the secrets model so that we can call on it to determine if destinations and associations should be called.

To test:
- On a fresh VRD, navigate to the Secrets sync page. You should not see any network requests made to the destinations or associations list endpoints. 
- Activate the feature via the opt-in modal.
- Look at the network request after the modal closes. The secrets model has been updated and thus the destinations and associations endpoints have been called.

---

<details>
<summary> Overview of commits </summary>

  - d56f564362241a25972f511194ce42ad7988c73c 

</details>
